Microsoft.HealthVault.Fhir/Transformers/MedicationToFhir.cs

```diff
@@ -0,0 +1,106 @@
+        {
+            return MedicationToFhir.ToFhirInternal(hvMedication, ToFhirInternal<MedicationStatement>(hvMedication));
+        }
+    }
+
+    internal static class MedicationToFhir
+    {
+        internal static MedicationStatement ToFhirInternal(HVMedication hvMedication, MedicationStatement medicationStatement)
+        {
+            var embeddedMedication = new FhirMedication();
+            string embeddedMedicationId = "med" + Guid.NewGuid();
+            embeddedMedication.Id = embeddedMedicationId;
+
+            medicationStatement.Contained.Add(ToFhirInternal(hvMedication, embeddedMedication));
+            medicationStatement.Medication = new ResourceReference(embeddedMedicationId);
+
+            medicationStatement.SetStatusAsActive();
+            medicationStatement.SetTakenAsNotApplicable();
+
+            medicationStatement.Dosage = AddDosage(hvMedication.Dose, hvMedication.Frequency, hvMedication.Route);
+
+            return medicationStatement;
+        }
+
+        private static List<Dosage> AddDosage(GeneralMeasurement dose, GeneralMeasurement frequency, CodableValue route)
+        {
+            return new List<Dosage> {
+                HealthVaultCodesToFhir.GetDosage(dose, frequency, route)
+                };
+        }
+
+        private static void SetStatusAsActive(this MedicationStatement medicationStatement)
+        {
+            medicationStatement.Status = MedicationStatement.MedicationStatementStatus.Active;
+        }
+
+        private static void SetTakenAsNotApplicable(this MedicationStatement medicationStatement)
+        {
+            medicationStatement.Taken = MedicationStatement.MedicationStatementTaken.Na;
+        }
+
+        internal static FhirMedication ToFhirInternal(HVMedication hvMedication, FhirMedication fhirMedication)
+        {
+            fhirMedication.Code = HealthVaultCodesToFhir.ConvertCodableValueToFhir(hvMedication.Name);
+
+            if (hvMedication.GenericName != null)
+            {
+                var ingredientItem = HealthVaultCodesToFhir.ConvertCodableValueToFhir(hvMedication.GenericName);
+                fhirMedication.Ingredient = new List<FhirMedication.IngredientComponent> {
+                    new FhirMedication.IngredientComponent(){
+                        Item= ingredientItem
+                }};
+            }
+
+            if (hvMedication.Strength != null)
+            {
+                if (!fhirMedication.Ingredient.Any())
+                {
+                    var ingredientItem = HealthVaultCodesToFhir.ConvertCodableValueToFhir(hvMedication.Name);
+                    fhirMedication.Ingredient = new List<FhirMedication.IngredientComponent> {
+                        new FhirMedication.IngredientComponent{
+                            Item = ingredientItem
+                    }};
+                }
+
+                var structuredMeasurement = hvMedication.Strength.Structured.First();
```

Comment:
It appears that structured can occur 0 times, so you'll want to null check this. https://docs.microsoft.com/en-us/healthvault/reference/datatypes/base#general-measurement